test/unit/logic/block.js

```diff
@@ -317,40 +326,65 @@ describe('block', () => {
 		done();
 	});
 
-	describe('with valid block and data', () => {
-		beforeEach(done => {
-			data = _.cloneDeep(validDataForBlock);
-			transactions = _.values(transactionsByTypes);
-			done();
+	beforeEach(done => {
+		data = _.cloneDeep(validDataForBlock);
+		transactions = _.values(transactionsByTypes);
+		done();
+	});
+
+	describe('create', () => {
+		var blockNormalizeStub;
+
+		before(() => {
+			blockNormalizeStub = sinonSandbox
+				.stub(block, 'objectNormalize')
+				.returnsArg(0);
+
+			transactionStub.getBytes.returns(Buffer.from('dummy transaction bytes'));
+			return transactionStub.objectNormalize.returnsArg(0);
 		});
 
-		describe('create', () => {
-			var blockNormalizeStub;
+		after(() => {
+			blockNormalizeStub.reset();
+			transactionStub.getBytes.reset();
+			return transactionStub.objectNormalize.reset();
+		});
 
-			before(() => {
-				blockNormalizeStub = sinonSandbox
-					.stub(block, 'objectNormalize')
-					.returnsArg(0);
+		describe('when one of all transaction types are present', () => {
+			var generatedBlock;
+			var transactionsOrder;
+			var correctOrder = [0, 1, 2, 3, 5, 6, 7, 4];
 
-				transactionStub.getBytes.returns(
-					Buffer.from('dummy transaction bytes')
-				);
-				return transactionStub.objectNormalize.returnsArg(0);
+			beforeEach(done => {
```

Comment:
`beforeEach` for one `it`?

Reply:
yes, I tried to update it with before hook and it doesn't work, so I have kept it as it is.

Reply:
ok, it's fine